File: lab_3/src/model/Producer.java
package model;

public class Producer extends Thread {
    private final Storage storage;
    private final int itemsToProduce;
    private final int id;

    public Producer(int id, Storage storage, int itemsToProduce) {
        this.id = id;
        this.storage = storage;
        this.itemsToProduce = itemsToProduce;
    }

    @Override
    public void run() {
        try {
            for (int i = 0; i < itemsToProduce; i++) {
                storage.produce(i, id);
                Thread.sleep(100);
            }
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}

File: lab_3_C#/Producer.cs
public class Producer
{
    private int id;
    private Storage storage;
    private int itemsToProduce;

    public Producer(int id, Storage storage, int itemsToProduce)
    {
        this.id = id;
        this.storage = storage;
        this.itemsToProduce = itemsToProduce;
    }

    public void Run()
    {
        for (int i = 0; i < itemsToProduce; i++)
        {
            storage.Produce(i, id);
            Thread.Sleep(100); 
        }
    }
}
